src/lints/struct_field_marked_deprecated.ron

```diff
@@ -30,16 +31,12 @@ SemverQuery(
                 item {
                     ... on Struct {
                         visibility_limit @filter(op: "=", value: ["$public"])
+                        deprecated @filter(op: "!=", value: ["$true"])
```

Comment:
Would you mind adding some comments around these new `deprecated` lines that explain why they are necessary and a good idea?

All the necessary info is in the PR, but we're not going to remember the PR in 3 months and it'll be harder to track that info the next time someone is wondering about it.